have a default configuration for sitemaps

dbu opened the pull request:
we now have a configuration section for defaults. and sitemaps are actually checked for valid configuration. 

if no configurations are specified, a `default` configuration with only the defaults will automatically exist. as soon as configurations are given, default has to be specified explicitly if it should exist. a configuration can be empty if its only used for specific voters or guessers but runs with the default templates + frequency.

dbu commented on DependencyInjection/CmfSeoExtension.php:
```diff
@@ -250,20 +251,34 @@ private function loadSitemapHandling($config, XmlFileLoader $loader, ContainerBu
+        $configurations = $config['configurations'];
         foreach ($config['configurations'] as $key => $configuration) {
-            $container->setParameter($this->getAlias().'.sitemap.'.$key.'_configuration', $configuration);
+//            $container->setParameter($this->getAlias().'.sitemap.'.$key.'_configuration', $configuration);
```
TODO: remove if really unused.

ElectricMaxxx replied:
thought that is the chance to create a default sitemap equal to the manager definition in the doctrine bundles.

dbu replied:
well, we are not using the parameter but only the configurations. i
prefer not to clutter the container with unused things. if the need for
it comes up, we can still add it at that point

ElectricMaxxx replied:
:+1:

dbu commented on Tests/Unit/DependencyInjection/CmfSeoExtensionTest.php:
```diff
@@ -272,6 +273,7 @@ public function testSitemapConfiguration()
             )
         );
 
+        /*
```
remove tests for removed configuration if no longer needed

dbu commented on DependencyInjection/CmfSeoExtension.php:
```diff
@@ -250,20 +251,34 @@ private function loadSitemapHandling($config, XmlFileLoader $loader, ContainerBu
+                    }
+                }
+            }
         }
```
TODO: create configuration 'default' if it was not customized, so that the controller can handle the default sitemap without special case.

dbu replied:
done

ElectricMaxxx replied:
Awesome.

dbu commented on DependencyInjection/Configuration.php:
```diff
@@ -106,12 +106,24 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->canBeEnabled()
                     ->children()
+                        ->arrayNode('defaults')
```
this is the real purpose of this PR: we can now configure defaults for all sitemap configurations in one place, and override them specifically.

dbu commented:
@ElectricMaxxx ok, updated and fixed. imho this would be ready. can you review and merge if you agree please? i can then squash the main PR and merge that. and then we have to start with doc or nobody will ever figure out how to use this :-)

ElectricMaxxx commented:
Travis seems to be unhappy. Will Review and merge tonight (of this afternoon). 
Yea documentation... There are more undocumented features for the SeoBundle atm: 
- alternate locale
- error pages
- sitemap

dbu commented:
the build issue was something about composer update failing. restarted those builds, lets see.

ElectricMaxxx commented:
thank @dbu

dbu commented:
alternate locales is noted in : https://github.com/symfony-cmf/symfony-cmf-docs/issues/582, error pages in https://github.com/symfony-cmf/symfony-cmf-docs/issues/606.

i now created https://github.com/symfony-cmf/symfony-cmf-docs/issues/681 for sitemap
